geopandas/testing.py

```diff
@@ -123,15 +123,28 @@ def assert_geoseries_equal(
     if not check_crs:
         with warnings.catch_warnings():
             warnings.filterwarnings("ignore", "CRS mismatch", UserWarning)
-            if check_less_precise:
-                assert geom_almost_equals(left, right)
-            else:
-                assert geom_equals(left, right)
+            check_equality(left, right, check_less_precise)
+    else:
+        check_equality(left, right, check_less_precise)
+
+
+def check_equality(left, right, check_less_precise):
+    if check_less_precise:
+        if not geom_almost_equals(left, right):
+            unequal_geoms = left[left.geom_almost_equals(right)]
+            raise AssertionError(
+                f"{len(unequal_geoms)} out of {len(left)} geometries"
+                f" are not almost equal. These geometries are "
+                f"not almost equal: {unequal_geoms}"
```

Comment:
```suggestion
                f"not almost equal: \n{unequal_geoms}"
```
Get the list of geometries starting on a new line.